### PR TITLE
[PM-221] Adding CipherId to the Send table, create/update sprocs, and added mi…

### PR DESCRIPTION
**Pull request by ttalty**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Adding the CipherId to the Send table and the create/update sprocs. 

## Code changes

* **src/Core/Tools/Entities/Send.cs:** Adding the CipherId to the entity
* **src/Sql/Tools/dbo/Stored Procedures/Send_Create.sql:** Adding the CipherId to the create sproc
* **src/Sql/Tools/dbo/Stored Procedures/Send_Update.sql:** Adding the CipherId to the update sproc
* **src/Sql/Tools/dbo/Tables/Send.sql:** CipherId added to the Send table
* **util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql:** Migrator script for sql changes
* **util/MySqlMigrations/Migrations/20240104162658_AddCipherIdToSend.cs:** MySql migration script
* **util/PostgresMigrations/Migrations/20240104162650_AddCipherIdToSend.cs:** Postgres migration script
* **util/SqliteMigrations/Migrations/20240104162642_AddCipherIdToSend.cs:** Sqlite Migration script

### CODE REVIEW

**Review by withinfocus — DISMISSED** [review]
Besides my one ⛏️ this looks good to me and I don't see why the validation is causing all these refreshes to occur.

File: util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+END
+GO
+
+CREATE VIEW [dbo].[SendView]
+AS
+SELECT
+    *
+FROM
+    [dbo].[Send]
+GO
```
Comment (Hinton) [inline util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql:23]: Should use `CREATE OR ALTER` syntax, https://contributing.bitwarden.com/contributing/code-style/sql#creating-or-modifying-a-view

File: util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    BEGIN
+        IF @Type = 1 --File
+        BEGIN
+            EXEC [dbo].[User_UpdateStorage] @UserId
+        END
+        EXEC [dbo].[User_BumpAccountRevisionDate] @UserId
+    END
+    -- TODO: OrganizationId bump?
+END
+GO
```
Comment (Hinton) [inline util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql:102]: Create or alter, https://contributing.bitwarden.com/contributing/code-style/sql/#creating-or-modifying-a-function-or-stored-procedure

File: util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    WHERE
+        [Id] = @Id
+
+    IF @UserId IS NOT NULL
+    BEGIN
+        EXEC [dbo].[User_BumpAccountRevisionDate] @UserId
+    END
+    -- TODO: OrganizationId bump?
+END
+GO
```
Comment (Hinton) [inline util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql:158]: Create or alter, https://contributing.bitwarden.com/contributing/code-style/sql/#creating-or-modifying-a-function-or-stored-procedure

File: util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql
Comment (Hinton) [inline util/Migrator/DbScripts/2023-01-04_00_Send_Add_CipherId.sql:1]: This migration is run out of order which causes the errors in the CI.

**Review by Hinton — COMMENTED** [review]
Some migration nits.

**Review by withinfocus — APPROVED** [review]
I pushed my way through this and found a couple nice finds so this is good to go now.

**Review by djsmith85 — APPROVED** [review]
Changes are looking good. Nice work @ttalty and thank you @withinfocus for helping out

**Pull request merged**